src/js/profile/mobile/widget/mobile/Listview.js

```diff
@@ -352,23 +376,37 @@
 				var self = this,
 					newElement = CoreListviewProto._build.call(self, element),
 					isChildListview = element &&
-						selectorUtils.getClosestByClass(element.parentElement, "ui-listview"),
-					canvas,
-					context;
+						selectorUtils.getClosestByClass(element.parentElement, "ui-listview");
 
 				self._isChildListview = isChildListview;
 
 				if (!isChildListview) {
-					canvas = document.createElement("canvas");
-					context = canvas.getContext("2d");
-					canvas.classList.add(classes.BACKGROUND_LAYER);
-					newElement.insertBefore(canvas, newElement.firstElementChild);
-					self._context = context;
+					self._addCanvas(newElement);
 				}
 
 				return newElement;
 			};
 
+			prototype._getCanvas = function () {
+				var self = this,
+					canvas;
+
+				// If reference to canvas has been changed.
+				// Developer can remove canvas from list, eg. by element.innerHTML = "<li>item</li>"
+				if (self._context) {
```

Comment:
shouldn't be 
(if self._context && !self.context.canvas) ?

Reply:
You probably mean:
`if self._context && self._context.canvas`

This additional condition has no impact on current code.